Illegal base64 character 2d error occurs if you submit select uuid()

Hi,

We upgraded prestosql to 312 but the error occurred.

* version
312

* query
```select uuid()```

* error log
```
presto:default> select uuid();
java.lang.RuntimeException: Error fetching next at http://host:8080/v1/statement/executing/20190530_035337_00358_5j6u4/x18968561a7bc4c0890e448ce2f6ae78a/0 returned an invalid response: JsonResponse{statusCode=200, statusMessage=OK, headers={content-type=[application/json], date=[Thu, 30 May 2019 03:53:37 GMT], vary=[Accept-Encoding, User-Agent], x-content-type-options=[nosniff]}, hasValue=false} [Error: {"id":"20190530_035337_00358_5j6u4","infoUri":"http://host:8080/ui/query.html?20190530_035337_00358_5j6u4","partialCancelUri":"http://host:8080/v1/stage/20190530_035337_00358_5j6u4.0","nextUri":"http://host:8080/v1/statement/executing/20190530_035337_00358_5j6u4/x18968561a7bc4c0890e448ce2f6ae78a/1","columns":[{"name":"_col0","type":"uuid","typeSignature":{"rawType":"uuid","typeArguments":[],"literalArguments":[],"arguments":[]}}],"data":[["8bbf4e4f-1e28-494e-bfc3-b1ccfb8e45d6"]],"stats":{"state":"RUNNING","queued":false,"scheduled":true,"nodes":1,"totalSplits":17,"queuedSplits":17,"runningSplits":0,"completedSplits":0,"cpuTimeMillis":0,"wallTimeMillis":0,"queuedTimeMillis":1,"elapsedTimeMillis":38,"processedRows":0,"processedBytes":0,"peakMemoryBytes":0,"spilledBytes":0,"rootStage":{"stageId":"0","state":"RUNNING","done":false,"nodes":1,"totalSplits":17,"queuedSplits":17,"runningSplits":0,"completedSplits":0,"cpuTimeMillis":0,"wallTimeMillis":0,"processedRows":0,"processedBytes":0,"subStages":[]},"progressPercentage":0.0},"warnings":[]}
]
	at io.prestosql.client.StatementClientV1.requestFailedException(StatementClientV1.java:463)
	at io.prestosql.client.StatementClientV1.advance(StatementClientV1.java:402)
	at io.prestosql.cli.StatusPrinter.printInitialStatusUpdates(StatusPrinter.java:131)
	at io.prestosql.cli.Query.renderQueryOutput(Query.java:151)
	at io.prestosql.cli.Query.renderOutput(Query.java:136)
	at io.prestosql.cli.Console.process(Console.java:337)
	at io.prestosql.cli.Console.runConsole(Console.java:264)
	at io.prestosql.cli.Console.run(Console.java:154)
	at io.prestosql.cli.Presto.main(Presto.java:31)
Caused by: java.lang.IllegalArgumentException: Unable to create class io.prestosql.client.QueryResults from JSON response:
[{"id":"20190530_035337_00358_5j6u4","infoUri":"http://host:8080/ui/query.html?20190530_035337_00358_5j6u4","partialCancelUri":"http://host:8080/v1/stage/20190530_035337_00358_5j6u4.0","nextUri":"http://host:8080/v1/statement/executing/20190530_035337_00358_5j6u4/x18968561a7bc4c0890e448ce2f6ae78a/1","columns":[{"name":"_col0","type":"uuid","typeSignature":{"rawType":"uuid","typeArguments":[],"literalArguments":[],"arguments":[]}}],"data":[["8bbf4e4f-1e28-494e-bfc3-b1ccfb8e45d6"]],"stats":{"state":"RUNNING","queued":false,"scheduled":true,"nodes":1,"totalSplits":17,"queuedSplits":17,"runningSplits":0,"completedSplits":0,"cpuTimeMillis":0,"wallTimeMillis":0,"queuedTimeMillis":1,"elapsedTimeMillis":38,"processedRows":0,"processedBytes":0,"peakMemoryBytes":0,"spilledBytes":0,"rootStage":{"stageId":"0","state":"RUNNING","done":false,"nodes":1,"totalSplits":17,"queuedSplits":17,"runningSplits":0,"completedSplits":0,"cpuTimeMillis":0,"wallTimeMillis":0,"processedRows":0,"processedBytes":0,"subStages":[]},"progressPercentage":0.0},"warnings":[]}
]
	at io.prestosql.client.JsonResponse.<init>(JsonResponse.java:70)
	at io.prestosql.client.JsonResponse.execute(JsonResponse.java:144)
	at io.prestosql.client.StatementClientV1.advance(StatementClientV1.java:388)
	... 7 more
Caused by: java.lang.IllegalArgumentException: Invalid JSON string for [simple type, class io.prestosql.client.QueryResults]
	at io.airlift.json.JsonCodec.fromJson(JsonCodec.java:137)
	at io.prestosql.client.JsonResponse.<init>(JsonResponse.java:67)
	... 9 more
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `io.prestosql.client.QueryResults`, problem: Illegal base64 character 2d
 at [Source: (String)"{"id":"20190530_035337_00358_5j6u4","infoUri":"http://host:8080/ui/query.html?20190530_035337_00358_5j6u4","partialCancelUri":"http://host:8080/v1/stage/20190530_035337_00358_5j6u4.0","nextUri":"http://host:8080/v1/statement/executing/20190530_035337_00358_5j6u4/x18968561a7bc4c0890e448ce2f6ae78a/1","columns":[{"name":"_col0","type":"uuid","typeSignature":{"rawType":"uuid","typeArguments":[],"literalArguments":[],"arguments":[]}}],"data":[["8bbf4e4f-"[truncated 600 chars]; line: 1, column: 1099]
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:67)
	at com.fasterxml.jackson.databind.DeserializationContext.instantiationException(DeserializationContext.java:1608)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.wrapAsJsonMappingException(StdValueInstantiator.java:484)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.rewrapCtorProblem(StdValueInstantiator.java:503)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:285)
	at com.fasterxml.jackson.databind.deser.ValueInstantiator.createFromObjectWith(ValueInstantiator.java:229)
	at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:195)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:488)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1287)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:326)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4013)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3042)
	at io.airlift.json.JsonCodec.fromJson(JsonCodec.java:134)
	... 10 more
Caused by: java.lang.IllegalArgumentException: Illegal base64 character 2d
	at java.base/java.util.Base64$Decoder.decode0(Base64.java:743)
	at java.base/java.util.Base64$Decoder.decode(Base64.java:535)
	at java.base/java.util.Base64$Decoder.decode(Base64.java:558)
	at io.prestosql.client.FixJsonDataUtils.fixValue(FixJsonDataUtils.java:179)
	at io.prestosql.client.FixJsonDataUtils.fixData(FixJsonDataUtils.java:73)
	at io.prestosql.client.QueryResults.<init>(QueryResults.java:69)
	at jdk.internal.reflect.GeneratedConstructorAccessor13.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.call(AnnotatedConstructor.java:124)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:283)
	... 19 more
Query is gone (server restarted?)
```

--- a/presto-client/src/main/java/io/prestosql/client/ClientStandardTypes.java
+++ b/presto-client/src/main/java/io/prestosql/client/ClientStandardTypes.java
@@ -41,6 +41,7 @@ public final class ClientStandardTypes
     public static final String MAP = "map";
     public static final String JSON = "json";
     public static final String IPADDRESS = "ipaddress";
+    public static final String UUID = "uuid";
     public static final String GEOMETRY = "Geometry";
     public static final String BING_TILE = "BingTile";
 

--- a/presto-client/src/main/java/io/prestosql/client/FixJsonDataUtils.java
+++ b/presto-client/src/main/java/io/prestosql/client/FixJsonDataUtils.java
@@ -47,6 +47,7 @@ import static io.prestosql.client.ClientStandardTypes.TIMESTAMP;
 import static io.prestosql.client.ClientStandardTypes.TIMESTAMP_WITH_TIME_ZONE;
 import static io.prestosql.client.ClientStandardTypes.TIME_WITH_TIME_ZONE;
 import static io.prestosql.client.ClientStandardTypes.TINYINT;
+import static io.prestosql.client.ClientStandardTypes.UUID;
 import static io.prestosql.client.ClientStandardTypes.VARCHAR;
 import static java.util.Collections.unmodifiableList;
 import static java.util.Objects.requireNonNull;
@@ -164,6 +165,7 @@ final class FixJsonDataUtils
             case INTERVAL_YEAR_TO_MONTH:
             case INTERVAL_DAY_TO_SECOND:
             case IPADDRESS:
+            case UUID:
             case DECIMAL:
             case CHAR:
             case GEOMETRY:

--- a/presto-client/src/test/java/io/prestosql/client/TestFixJsonDataUtils.java
+++ b/presto-client/src/test/java/io/prestosql/client/TestFixJsonDataUtils.java
@@ -56,6 +56,7 @@ public class TestFixJsonDataUtils
         assertQueryResult("map(bigint,bigint)", ImmutableMap.of(1, 3, 2, 4), ImmutableMap.of(1L, 3L, 2L, 4L));
         assertQueryResult("json", "{\"json\": {\"a\": 1}}", "{\"json\": {\"a\": 1}}");
         assertQueryResult("ipaddress", "1.2.3.4", "1.2.3.4");
+        assertQueryResult("uuid", "0397e63b-2b78-4b7b-9c87-e085fa225dd8", "0397e63b-2b78-4b7b-9c87-e085fa225dd8");
         assertQueryResult("Geometry", "POINT (1.2 3.4)", "POINT (1.2 3.4)");
         assertQueryResult("map(BingTile,bigint)", ImmutableMap.of("BingTile{x=1, y=2, zoom_level=10}", 1), ImmutableMap.of("BingTile{x=1, y=2, zoom_level=10}", 1L));
     }

--- a/presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcResultSet.java
+++ b/presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcResultSet.java
@@ -118,6 +118,8 @@ public class TestJdbcResultSet
         checkRepresentation("cast('foo' as char(5))", Types.CHAR, "foo  ");
         checkRepresentation("ARRAY[1, 2]", Types.ARRAY, (rs, column) -> assertEquals(rs.getArray(column).getArray(), new int[] {1, 2}));
         checkRepresentation("DECIMAL '0.1'", Types.DECIMAL, new BigDecimal("0.1"));
+        checkRepresentation("IPADDRESS '1.2.3.4'", Types.JAVA_OBJECT, "1.2.3.4");
+        checkRepresentation("UUID '0397e63b-2b78-4b7b-9c87-e085fa225dd8'", Types.JAVA_OBJECT, "0397e63b-2b78-4b7b-9c87-e085fa225dd8");
 
         checkRepresentation("DATE '2018-02-13'", Types.DATE, (rs, column) -> {
             assertEquals(rs.getObject(column), Date.valueOf(LocalDate.of(2018, 2, 13)));

